zkevm-circuits/src/evm_circuit/execution/create.rs

```diff
@@ -177,23 +185,40 @@ impl<F: Field> ExecutionGadget<F> for CreateGadget<F> {
             );
         });
 
-        let transfer = TransferGadget::construct(
-            cb,
-            from_bytes::expr(&caller_address.cells),
+        // TODO：check for address collision case by code hash previous
+        cb.account_read(
             new_address.clone(),
-            0.expr(),
-            1.expr(),
-            value.clone(),
-            &mut callee_reversion_info,
+            AccountFieldTag::CodeHash,
+            code_hash_previous.expr(),
         );
 
-        cb.account_write(
-            new_address.clone(),
-            AccountFieldTag::Nonce,
-            1.expr(),
-            0.expr(),
-            Some(&mut callee_reversion_info),
+        let not_address_collision = IsZeroGadget::construct(cb, code_hash_previous.expr());
+        cb.require_zero(
+            "is_code_hash_previous_zero * code_hash_previous = 0",
+            code_hash_previous.expr() * not_address_collision.expr(),
         );
+        // TODO：conditional transfer for address collision case
```

Comment:
Remove todo？

Reply:
yes, will remove it. Thanks !

Reply:
removed in [52be462](https://github.com/scroll-tech/zkevm-circuits/pull/406/commits/52be46292621459896972f3cac9d66df191894a9) thanks!